synfig-core/src/synfig/string_helper.cpp

```diff
@@ -59,7 +59,7 @@ get_locale_decimal_point()
 }
 
 std::string
-synfig::remove_trailing_zeroes(const std::string& text, bool force_decimal_point)
+synfig::remove_trailing_zeroes(const std::string& text, bool force_decimal_point, bool has_unit)
```

Comment:
yes for any string we send which has a unit at the end, the original implementation didn't account for it. I mean we could parse the string to remove the unit before sending and then re-add it (but it may become redundant), also I feel this may be good as everywhere I seen it the unit was part of the string. 

if there's something i'm not noticing here though please let me know :)

Reply:
> if there's something i'm not noticing here though please let me know :)

As I see it in the current code - all values are floating point/double converted to string, so they don't have a trailing part.

Reply:
> > 
> so they don't have a trailing part.

How ? 
example: in `cell_renderervalue.cpp`
`x.get_string(real_num_decimals).c_str();` yields out trailing zeroes in the string which have to be removed using the `remove_trailing_zeroes()` method -and also the has_unit parameter has to be set to true for it to correct remove the zeroes-. And this is so the case for the value of type real which doesnt have a unit and all the other float to string types.

Reply:
> example: in `cell_renderervalue.cpp` line 341

```
Distance x( data.get(Real()), Distance::SYSTEM_UNITS);
x.convert( App::distance_system, get_canvas()->rend_desc() );
property_text() = x.get_string(real_num_decimals,true,true,true).c_str();
```

Distance contains float value without units.
```
Distance::get_string(int digits, bool remove_trailing_zeroes)const
{
        ...
	return strprintf("%s%s",str.c_str(),system_name(system_).c_str()); <- units added here
}
```
so trailing zeroes should be already removed.

Reply:
oh ok now I get it you mean we should remove the trailing zeroes before the units are even added which in turn removes the need for has_unit. will do.

Reply:
done